### PR TITLE
[MOB-19239] Update feed retrieval from remote

### PR DESCRIPTION
**Pull request by rymorale**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by swarna04 — CHANGES_REQUESTED** [review]

File: code/messaging/src/main/java/com/adobe/marketing/mobile/Surface.java [hunk 1]
```diff
@@ -54,4 +62,9 @@ public boolean isValid() {
 
         return this.uri.startsWith(SURFACE_BASE);
     }
+
+    public static Surface fromString(final String path) {
```
[the remark this answers is not in the document]
Comment (rymorale) [reply code/messaging/src/main/java/com/adobe/marketing/mobile/Surface.java:66]: i need to keep it public as its called from MessagingUtils: https://github.com/rymorale/aepsdk-messaging-android/blob/update-feed-retrieval-from-remote/code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/internal/MessagingUtils.java#L161-L172

File: code/messaging/src/main/java/com/adobe/marketing/mobile/Surface.java [hunk 2]
```diff
@@ -31,13 +31,21 @@ public class Surface {
...
     public Surface(final String path) {
-        final String packageName = ServiceProvider.getInstance().getDeviceInfoService().getApplicationPackageName();
-        final String baseUri = StringUtils.isNullOrEmpty(packageName) ? UNKNOWN_SURFACE : SURFACE_BASE + packageName;
-        this.uri = StringUtils.isNullOrEmpty(path) || baseUri.equals(UNKNOWN_SURFACE) ? baseUri : baseUri + File.separator + path;
+        this(false, path);
     }
 
     public Surface() {
-        this(null);
+        this(false, null);
```
[the remark this answers is not in the document]
Comment (rymorale) [reply code/messaging/src/main/java/com/adobe/marketing/mobile/Surface.java:38]: yeah I think that is a better way to do it. will update it.

File: code/messaging/src/main/java/com/adobe/marketing/mobile/Surface.java [hunk 3]
```diff
@@ -54,4 +61,9 @@ public boolean isValid() {
 
         return this.uri.startsWith(SURFACE_BASE);
     }
+
+    public static Surface fromUriString(final String path) {
```
Comment (swarna04) [inline code/messaging/src/main/java/com/adobe/marketing/mobile/Surface.java:65]: As per previous feedback, also requested renaming path->uri

File: code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/internal/MessagingUtils.java [hunk 4]
```diff
@@ -132,36 +133,46 @@ static boolean isEdgePersonalizationDecisionEvent(final Event event) {
...
-    static List<String> getSurfaces(final Event event) {
+    static List<Surface> getSurfaces(final Event event) {
         if (event == null || event.getEventData() == null) {
             return null;
         }
-        return DataReader.optTypedList(String.class, event.getEventData(), MessagingConstants.EventDataKeys.Messaging.SURFACES, null);
+        final List<String> surfaceUris = DataReader.optTypedList(String.class, event.getEventData(), MessagingConstants.EventDataKeys.Messaging.SURFACES, null);
+
+        if (surfaceUris == null || surfaceUris.isEmpty()) {
+            return null;
```
Comment (swarna04) [inline code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/internal/MessagingUtils.java:168]: nit: Let us provide a debug log here before returning null.

**Review by swarna04 — APPROVED** [review]

**Pull request merged**